---
title: "Invasional meltdown: soil"
author: "Zhijie_Zhang"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  html_document:
    code_folding: hide
    toc: true
    toc_depth: 3
    number_sections: true
    toc_float:
      collapsed: true
      smooth_scroll: no
---


<style type="text/css">

body{ /* Normal  */
      font-size: 16px;
      font-family: "Times New Roman"
  }
td {  /* Table  */
  font-size: 14px;
}
h1.title {
  font-size: 34px;
  color: Black;
  font-family: "Arial";
}
h1 { /* Header 1 */
  font-size: 26px;
  color: DarkBlue;
  font-family: "Arial";
}
h2 { /* Header 2 */
    font-size: 22px;
    font-family: "Arial";
    color: DarkBlue;
}
h3 { /* Header 3 */
  font-size: 18px;
  font-family: "Arial";
  color: DarkBlue;
}

h4 { /* Header 4 */
  font-size: 16px;
  font-family: "Arial";
  color: DarkBlue;
}
code.r{ /* Code block */
    font-size: 12px;
}
pre { /* Code block - determines code spacing between lines */
    font-size: 12px;
}
</style>


The codes for the paper **Soil-microbes-mediated invasional meltdown in plants**. This is the analyses for soil-microbe data.\


**Please run [Invasional meltdown: plant] first.**\

Plant data [**Invasional meltdown: plant**] showed that aliens grew better than natives on soil conditioned by aliens. On soil that was not conditioned or conditioned by natives, aliens and natives were similar.\

Here I analyzed soil samples from the soil-conditioning phase (4 alien + 6 native species + non-conditioned soil, sterilized or not before training. I have 115 samples for bacterial communities and 111 samples for fungal communities.\

Note that, when testing the relationship (beta, $\beta$ in Fig. 1) between soil-legacy effect and soil-communities (alpha, $\alpha$ in Fig. 1), I only used non-sterilized soil. This is because the test plants were only grown on the non-sterilized soil.


**Findings**:


1. Alien and native conditioned the soil in different ways, reflected by compositions of baterial and fungal communities (section 3).
2. Aliens and aliens shared a few fungal endophytes. Alien and natives shared some endophytes And native and natives shared many endophytes (section 4.1).
3. Alpha divesrity of soil microbes and relative abundance of fungal pathogens, overally, did not differ between aliens and natives. (section 5).
4. Soil-legacy effect was not significantly explained by diversity/abundance of soil microbes (section 6.2), but was significantly explained by the similarity in fungal endophyte communities between plant species (section 6.1.1).


```{r}
knitr::opts_chunk$set(warning=F,message = F)
dir.create('table',showWarnings = F)
dir.create('fig',showWarnings = F)
```


```{r}
source('01functions.R')
load('04phyloseq_16s_its.RData') # phyloseq, cleaned
load('legacy_beta.RData')      # soil legacy effect, from plant data.
```

```{r,message=F,warning=F}
library(stringr)
library(corrplot)
library(nlme)
library(vegan)
library(phyloseq)
library(themetagenomics)
require(reshape2)
library(RColorBrewer)
library(tidyverse)
library(gridExtra)
library(phytools)
library(knitr)
library(shape)
library(pBrackets)
library(scales)
library(MuMIn)
```


```{r}
col_fig <- brewer.pal(3, 'Dark2')[2:1]
theme <- theme_bw() + 
  theme(panel.background = element_blank(), panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), 
        #axis.line = element_line(colour = "black"),
        strip.background = element_blank(),
        legend.key = element_rect(colour = NA, fill = NA),
        legend.background = element_blank(),
        legend.title = element_blank(),
        axis.text.x = element_text(size = 6, colour = 'black'),
        axis.text.y = element_text(size = 6, colour = 'black'),
        axis.title  = element_text(size = 6, colour = 'black'),
        title = element_text(size = 7),
        #legend.position = "none",
        plot.title = element_text(size = 7, hjust = 0.5),
        plot.tag   = element_text(size = 7, face = 'bold'),
        legend.position = "none",
        plot.margin = margin(t = 0.5, r = 0.1, b = 0.5, l = 0.1, unit = "cm"))
```




# data preparation


check reads and decide the rarefy depth
```{r}
print('samples with the lowest reads')
(sort(sample_sums(phyloseq_16s_raw)))[1:5]
(sort(sample_sums(phyloseq_its_raw)))[1:5]
```


rarefy 16s and its to same reads, 30000 for 16s, and 9500 for its\
one samples (1626) were removed for 16s\
two samples (1770 and 1767) were removed for its.

```{r, include = F}
set.seed(100)
phyloseq_16s <- rarefy_even_depth(phyloseq_16s_raw,sample.size = 30000,
                          rngseed = FALSE, replace = TRUE, trimOTUs = TRUE, verbose = TRUE)
set.seed(100)
phyloseq_its <- rarefy_even_depth(phyloseq_its_raw,sample.size = 9500,
                                  rngseed = FALSE, replace = TRUE, trimOTUs = TRUE, verbose = TRUE)
rm(phyloseq_its_raw, phyloseq_16s_raw)
```


get otu/tax/sample tables
```{r}
# 16s
otu_table_16s <- as.data.frame(phyloseq_16s@otu_table@.Data)
tax_table_16s <- as.data.frame(phyloseq_16s@tax_table@.Data)
sam_table_16s <- as(sample_data(phyloseq_16s), "data.frame") %>% 
  # -------------- rename some variables to make them consistent with the plant data ---------------#
  mutate(status = recode(status, 'a' = 'alien', 'n' = 'native', 'ck' = 'non-conditioned')) %>% 
  mutate(species = recode(species, 'Solidago' = "Solidago_gigantea",  'Leontodon' = "Leontodon_autumnalis",
                          'Salvia_ver' ="Salvia_verticillata", 'Salvia' = "Salvia_pratensis",
                          'Plantago' = "Plantago_media", 'Onobrychis' =  "Onobrychis_viciifolia",
                          'Lotus' = "Lotus_corniculatus", 'Lolium' = "Lolium_multiflorum", 
                          'Cynosurus' = "Cynosurus_cristatus", 'Dactylis' = "Dactylis_glomerata", 
                          'ck' = 'non-conditioned')) %>% 
  arrange(match(sample_id, rownames(otu_table_16s)))

# its
otu_table_its <- as.data.frame(phyloseq_its@otu_table@.Data)
tax_table_its <- as.data.frame(phyloseq_its@tax_table@.Data)
sam_table_its <- as(sample_data(phyloseq_its), "data.frame") %>%
  mutate(status = recode(status, 'a' = 'alien', 'n' = 'native', 'ck' = 'non-conditioned')) %>% 
  mutate(species = recode(species,'Solidago' = "Solidago_gigantea", 
                          'Leontodon' = "Leontodon_autumnalis", 
                          'Salvia_ver' ="Salvia_verticillata",
                          'Salvia' = "Salvia_pratensis", 
                          'Plantago' = "Plantago_media", 
                          'Onobrychis' =  "Onobrychis_viciifolia",
                          'Lotus' = "Lotus_corniculatus", 
                          'Lolium' = "Lolium_multiflorum", 
                          'Cynosurus' = "Cynosurus_cristatus", 
                          'Dactylis' = "Dactylis_glomerata", 
                          'ck' = 'non-conditioned')) %>% 
  arrange(match(sample_id, rownames(otu_table_its)))
info_family<- data.frame(species = sort(unique(sam_table_16s$species)),
                         family = c('non-conditioned','Poaceae','Poaceae','Compositae','Poaceae',
                                    'Leguminosae','Leguminosae','Plantaginaceae','Lamiaceae','Lamiaceae','Compositae')) #informations

sam_table_16s <- sam_table_16s%>%
  left_join(info_family) %>% 
  mutate(T1_empty = ifelse(status == 'non-conditioned', -2/3, 1/3),
         T2_alien_native = ifelse(status == 'non-conditioned', 0,
                                  ifelse(status == 'alien', 0.5, -0.5))) %>% 
  mutate(trans_date_c=ifelse(fix %in% c(1360:1400), 28,# add transplanting date, salvia
                             ifelse(fix %in% c(1492:1568), 33, # solidago gigantea
                                    ifelse(species == "Salvia_verticillata", 15,
                                           ifelse(species == "Solidago_gigantea", 20,
                                                  ifelse(species =='Plantago_media', 2,0))))))
sam_table_its <- sam_table_its %>%
  left_join(info_family) %>% 
  # set contrast for the three levels (non-conditioned, alien, native)
  mutate(T1_empty = ifelse(status == 'non-conditioned', -2/3, 1/3),
         T2_alien_native = ifelse(status == 'non-conditioned', 0,
                                  ifelse(status == 'alien', 0.5, -0.5))) %>% 
  # ------------add transplanting date-------------#
  mutate(trans_date_c = ifelse(fix %in% c(1360:1400), 28, # salvia
                               ifelse(fix %in% c(1492:1568), 33,# solidago
                                      ifelse(species == "Salvia_verticillata", 15,
                                             ifelse(species == "Solidago_gigantea", 20,
                                                    ifelse(species == 'Plantago_media', 2, 0))))))

#change order of status, make the plotting easier
sam_table_its$status <- factor(sam_table_its$status, levels = c('non-conditioned', 'alien', 'native'))
sam_table_16s$status <- factor(sam_table_16s$status, levels = c('non-conditioned', 'alien', 'native'))
```




# assign its to FUNGuild 
get the names of the taxa, and then match functional groups from the FUNGuild.\
accessed on 2019-10-30
```{r, eval=F}

# its_otu<- as.data.frame(phyloseq_its_raw@otu_table)
# transposed_otu_table <- as.data.frame(t(its_otu))# Table transposition
# transposed_otu_table$OTU_ID <- rownames(transposed_otu_table)
# transposed_otu_table<-remove_rownames(transposed_otu_table)
# 
# 
# its_tax<-as.data.frame(phyloseq_its_raw@tax_table@.Data)
# tax <- data.frame(taxonomy=paste(its_tax$Kingdom,
#                                  its_tax$Phylum,
#                                  its_tax$Class,
#                                  its_tax$Order,
#                                  its_tax$Family,
#                                  its_tax$Genus,
#                                  its_tax$Species, sep=";"),its_tax[,0])
# tax$OTU_ID <- rownames(tax)
# tax<-remove_rownames(tax)
# 
# df_its_tax<-transposed_otu_table%>%
#   left_join(tax)
# 
# write.table(df_its_tax, file = paste(path,'dfr_funguild.txt', sep = ''),
#             quote=FALSE, sep = " ", dec = ".",
#             row.names = FALSE, col.names = TRUE)
# rm(df_its_tax,tax,its_tax,its_otu,transposed_otu_table)
# Run funguil_assigment on http://www.stbates.org/guilds/app.php. Register the file in csv df_funguilds.guilds.txt

# Formatting its tables for complete statistical analysis (samples in rows and variables in columns)
```




# soil-communitiy composition
**This section aims to test whether plant affected the soil communities, and whether the effect depends on the origin (native vs alien).**

the contrasts:\
T1_empthy: ck vs mean of native and alien\
T2 alien_native: alien vs native\

data prep
```{r}
# filter data
sam_table_16s_nonster <- sam_table_16s %>%
  filter(ster.not == 'nonster') %>% droplevels()

sam_table_16s_ster <- sam_table_16s %>%
  filter(ster.not == 'ster') %>% droplevels()

sam_table_its_nonster <- sam_table_its %>%
  filter(ster.not == 'nonster') %>% droplevels()

sam_table_its_ster   <- sam_table_its %>%
  filter(ster.not=='ster') %>% droplevels()


# extract otu
otu_table_16s_nonster <- otu_table_16s[as.character(sam_table_16s_nonster$sample_id),]
otu_table_16s_ster    <- otu_table_16s[as.character(sam_table_16s_ster$sample_id),]
otu_table_its_nonster <- otu_table_its[as.character(sam_table_its_nonster$sample_id),]
otu_table_its_ster    <- otu_table_its[as.character(sam_table_its_ster$sample_id),]

```

assign functional groups
```{r}
its_func <- read.csv('05dfr_funguild_guilds.csv')
# clean the functional groups
its_func <- its_func %>% 
  mutate(pathogen = ifelse(grepl('plant patho|plant parasite', Guild, ignore.case = T) &
                             !(grepl('polen',Guild,ignore.case = T)),'pathogen',0), # some are pollen (polen is a typo in the funguild?) 
         AMF      = ifelse(grepl('Arbuscular Mycorrhizal', Guild, ignore.case = T), 'AMF', 0),
         endo     = ifelse(grepl('endophyt',               Guild, ignore.case = T), 'endo',0),
         assigned = ifelse(Guild =='-', 0, 'assigned')) %>% 
  mutate(endo_path    = ifelse(endo == 'endo' & pathogen == 'pathogen', 'endo_path', 0),    # pathogenic endophytes
         endo_no_path = ifelse(endo == 'endo' & pathogen != 'pathogen', 'endo_no_path', 0), # non-pathogenic endophytes
         )
```

```{r}
its_func          <- its_func %>% filter(OTU_ID %in% as.character(colnames(otu_table_its)))
its_func_nonster  <- its_func %>% filter(OTU_ID %in% as.character(colnames(otu_table_its_nonster)))
its_func_ster     <- its_func %>% filter(OTU_ID %in% as.character(colnames(otu_table_its_ster)))
```


```{r}
# otu table, subset
# -------------not sterilized----------------#
otu_table_its_nonster_pathogen <-
  otu_table_its_nonster[,as.character(its_func_nonster$OTU_ID[its_func_nonster[,'pathogen'] == 'pathogen'])] # pathogen
otu_table_its_nonster_endo <- 
  otu_table_its_nonster[,as.character(its_func_nonster$OTU_ID[its_func_nonster[,'endo'] == 'endo'])]            # endophytes

# two subsets of endophytes
otu_table_its_nonster_endo_path <-
  otu_table_its_nonster[,as.character(its_func_nonster$OTU_ID[its_func_nonster[,'endo_path'] == 'endo_path'])]
otu_table_its_nonster_endo_no_path <-
  otu_table_its_nonster[,as.character(its_func_nonster$OTU_ID[its_func_nonster[,'endo_no_path'] == 'endo_no_path'])]

otu_table_its_nonster_amf  <- 
  otu_table_its_nonster[,as.character(its_func_nonster$OTU_ID[its_func_nonster[,'AMF'] == 'AMF'])]               # AMF

# ---------------sterilized soil ----------------#
otu_table_its_ster_pathogen <- otu_table_its_ster[,as.character(its_func_ster$OTU_ID[its_func_ster[,'pathogen'] == 'pathogen'])]
otu_table_its_ster_endo     <- otu_table_its_ster[,as.character(its_func_ster$OTU_ID[its_func_ster[,'endo'] == 'endo'])]
otu_table_its_ster_amf      <- otu_table_its_ster[,as.character(its_func_ster$OTU_ID[its_func_ster[,'AMF'] == 'AMF'])]

```



NDMS: 16s (bacteria), its (fungi) and fungal pathogens.\
Bray-Curtis dissimilarity was calculated between samples.
```{r, message=FALSE}
beta_all <- lapply(list(otu_table_16s_nonster,      otu_table_16s_ster, # bacteria
                    otu_table_its_nonster,          otu_table_its_ster, # fungi
                    otu_table_its_nonster_pathogen, otu_table_its_ster_pathogen), # fungal pathogens
               function(otu) {metaMDS(otu, trymax = 10000, autotransform = FALSE, trace = FALSE, k=2)})

```


```{r}
# colors for the NDMS figures
col_table <- sam_table_16s_nonster %>%
  dplyr::select(status,species) %>%
  distinct() %>%
  arrange(status) %>%
  mutate(pch = 16) # order according to status
col_table$col<-c('black',#
                 brewer.pal(n = 9, name = "Oranges")[c(4:7)],
                 brewer.pal(n = 9, name = "Greens")[c(3:8)]
                 )
# info
sam_table_16s_nonster <- sam_table_16s_nonster %>% left_join(col_table)
sam_table_16s_ster    <- sam_table_16s_ster    %>% left_join(col_table)
sam_table_its_nonster <- sam_table_its_nonster %>% left_join(col_table)
sam_table_its_ster    <- sam_table_its_ster    %>% left_join(col_table)
```


all models\
Note that as adonis is tested with randomization, the results in the report will differ sightly with that in the manuscript.
```{r}

# -------------non sterilized----------------#
t_struct_16s <- adonis(otu_table_16s_nonster~ T1_empty + T2_alien_native + family + species + trans_date_c, 
                       data = sam_table_16s_nonster)
t_struct_its <- adonis(otu_table_its_nonster ~ T1_empty + T2_alien_native + family + species + trans_date_c,
                       data = sam_table_its_nonster)
t_struct_its_nonster_pathogen <- adonis(otu_table_its_nonster_pathogen ~ T1_empty + T2_alien_native + 
                                          family + species + trans_date_c, 
                                        data = sam_table_its_nonster)

# -------------sterilized----------------#
t_struct_16s_ster <- adonis(otu_table_16s_ster~ T1_empty + T2_alien_native + family + species + trans_date_c,
       data = sam_table_16s_ster)

t_struct_its_ster <- adonis(otu_table_its_ster ~ T1_empty + T2_alien_native + family + species + trans_date_c,
                            data = sam_table_its_ster)
t_struct_its_ster_pathogen    <- adonis(otu_table_its_ster_pathogen ~ T1_empty + T2_alien_native + 
                                          family + species + trans_date_c, 
                                        data = sam_table_its_ster)
#-------------------------#
t_struct_16s <- t_struct_16s$aov.tab %>% as.data.frame()
t_struct_its <- t_struct_its$aov.tab %>% as.data.frame()
t_struct_its_nonster_pathogen <- t_struct_its_nonster_pathogen$aov.tab %>% as.data.frame()

t_struct_16s_ster <- t_struct_16s_ster$aov.tab %>% as.data.frame()
t_struct_its_ster <- t_struct_its_ster$aov.tab %>% as.data.frame()
t_struct_its_ster_pathogen <- t_struct_its_ster_pathogen$aov.tab %>% as.data.frame()
```


## non-sterilized soil
### 16s (bacteria)
```{r}
t_struct_16s %>% kable(digits = 3)
```

### its (all fungi)
```{r}
t_struct_its %>% kable(digits = 3)
```

### fungal pathogen
```{r}
t_struct_its_nonster_pathogen %>% kable(digits = 3)
```


## sterilized
### 16s (bacteria)
```{r}
t_struct_16s_ster %>% kable(digits = 3)
```

### its (all fungi)
```{r}
t_struct_its_ster %>% kable(digits = 3)
```

### fungal pathogens
```{r}
t_struct_its_ster_pathogen %>% kable(digits = 3)
```


## figures
### bacteria and fungi (Extended Fig. 2)
```{r, fig.height=6, fig.width=7}
# pdf('./fig/struct.pdf', height = 6, width = 18/2.54)
par(mfrow = c(2,2), xpd = T)

beta_plot(beta_all[[1]],sam_table_16s_nonster,ylim = c(-0.2,0.2),xlim = c(-0.4,1), 
          yaxt = c(-0.2, -0.1, 0, 0.1, 0.2), xaxt = c(-0.4,0,0.4,0.8))
title('Bacteria (Live)', cex.main = 0.8, font.main = 1)


beta_plot(beta_all[[2]], sam_table_16s_ster,xlim = c(-1,1),ylim = c(-1,0.6),
          yaxt = c(-0.8,-0.4,0,0.4), xaxt = c(-1,-0.5,0, 0.5, 1), tag = 'b')
title('Bacteria (Sterilized)', cex.main = 0.8, font.main = 1)
legend(1.05,0.6, col_table$species, col = col_table$col, pch = 16,bty = 'n', cex = 0.6)

beta_plot(beta_all[[3]], sam_table_its_nonster,xlim = c(-1.2,2.2),ylim = c(-1.8,2),
          yaxt = c(-1, 0, 1, 2), tag = 'c')
title('Fungi (Live)', cex.main = 0.8, font.main = 1)

beta_plot(beta_all[[4]], sam_table_its_ster,xlim = c(-1.7,2.2),ylim = c(-1.8,2.1),
          yaxt = NULL, xaxt = c(-1, 0, 1, 2), tag = 'd')
title('Fungi (Sterilized)', cex.main = 0.8, font.main = 1)
# dev.off()
```

### fungal pathogens (Supplementary Figure 3)
```{r, fig.height= 3, fig.width=7}
# pdf('./fig/struct_fungi.pdf', height = 3, width = 18/2.54)
par(mfrow = c(1,2), xpd = T)
beta_plot(beta_all[[5]],sam_table_its_nonster,
          ylim = c(-1.8,1.5), xlim = c(-1.5,1.5), 
          yaxt = c(-1, 0, 1), xaxt = c(-1, 0, 1))
title('Fungal pathogens (Live)', cex.main = 0.8, font.main = 1)

beta_plot(beta_all[[6]],sam_table_its_ster,
          ylim = c(-2,2.2), xlim = c(-2.5,2.5), 
          yaxt = c(-2, -1, 0, 1, 2), xaxt = c(-2, 0, 2), tag = 'b')
title('Fungal pathogens (Sterilized)', cex.main = 0.8, font.main = 1)
legend(2.7,2, col_table$species, col = col_table$col, pch = 16, bty = 'n', cex = 0.6)

# dev.off()
```



export tables
```{r}
# write.csv(rbind(t_struct_16s,t_struct_16s_ster,
#                 t_struct_its, t_struct_its_ster), 
#           './table/t_structure.csv')
```

```{r}
# write.csv(rbind(t_struct_its_nonster_pathogen, t_struct_its_ster_pathogen), './table/t_structure_pathogen.csv')
```


# soil communities dissimilarity between plants
**This section aims to test whether similarity of soil communities (Bray-Curtis dissimilarities, or beta diversity) depends on the combination of species.** (five combinations: alien on its own [intra_a], native on its own [intra_n],  aliens with aliens[a a], aliens with natives[n a], and natives with natives[n n])\

I first calculated the beta diversity between every two samples.\
Then, basing on them, I calculated the mean beta diversity  between every two-species combinations.\
Last, I tested whether the five combinations (intra_a, intra_n, aa, na, nn) matters.


```{r}
info_16s <- sam_table_16s %>%
  dplyr::select(species, status, ster.not, sample_id, trans_date_c) %>% 
  arrange(ster.not, status, species, sample_id) %>% 
  droplevels()

info_its <- sam_table_its %>%
  dplyr::select(species, status, ster.not, sample_id, trans_date_c)
```



**aa** means alien with alien, **na** means native with alien, and **nn** means native with native.
```{r}
beta_mat_sp_16s_nonster <- beta_by_sp(otu_table_16s_nonster, info_16s) # matrix for 16s nonsterilized
beta_mat_sp_its_nonster <- beta_by_sp(otu_table_its_nonster, info_its) # its, nonsterilized
beta_mat_sp_16s_ster    <- beta_by_sp(otu_table_16s_ster,    info_16s) # 16s, sterilized
beta_mat_sp_its_ster    <- beta_by_sp(otu_table_its_ster,    info_its) # its, sterilized
```
```{r}
beta_mat_sp_its_nonster_pathogen <- beta_by_sp(otu_table_its_nonster_pathogen, info_its)
beta_mat_sp_its_nonster_endo     <- beta_by_sp(otu_table_its_nonster_endo,     info_its)
beta_mat_sp_its_nonster_endo_path    <- beta_by_sp(otu_table_its_nonster_endo_path,   info_its)
beta_mat_sp_its_nonster_endo_no_path <- beta_by_sp(otu_table_its_nonster_endo_no_path,info_its)

beta_mat_sp_its_ster_pathogen <- beta_by_sp(otu_table_its_ster_pathogen,info_its)
beta_mat_sp_its_ster_endo     <- beta_by_sp(otu_table_its_ster_endo,    info_its)
```


## non-sterilized soil
### 16s (bacteria)
```{r}
u<-gl(1,1,length(beta_mat_sp_16s_nonster$mean_beta))
model <- lme(logit(mean_beta)~ aa_vs_na + na_vs_nn + intra_vs_inter + intra_a_vs_n, 
             random=list(u=pdBlocked(list(
              pdIdent(form=~sp1-1),
              pdIdent(form=~sp2-1),
              pdIdent(form=~family1-1),
              pdIdent(form=~family2-1)
              ))),
             control = lmeControl(msMaxIter = 10000),
             weights = varIdent(form = ~1|combi_status),
             data = beta_mat_sp_16s_nonster, method = 'ML')
```

```{r}
t_beta_16s <- merge_fix_rand(drop1(model, test = 'Chisq')[-1,], model) 
t_beta_16s %>% kable
```

### its (fungi)
```{r}
print('all fungi')
u <- gl(1,1,length(beta_mat_sp_its_nonster$mean_beta))
model <- update(model, weights = varIdent(form = ~1|combi_status), data = beta_mat_sp_its_nonster)
t_beta_its <- merge_fix_rand(drop1(model, test = 'Chisq')[-1,], model)
t_beta_its %>% kable
```

```{r}
print('pathogen')
u<-gl(1,1,length(beta_mat_sp_its_nonster_pathogen$mean_beta))
model <- update(model, weights = varIdent(form = ~1|combi_status), 
                data = beta_mat_sp_its_nonster_pathogen,
                control = lmeControl(opt = 'optim'))

t_beta_its_path <- merge_fix_rand(drop1(model, test = 'Chisq')[-1,], model) 
t_beta_its_path %>% kable
```

```{r}
print('all endophytes')
u<-gl(1,1,length(beta_mat_sp_its_nonster_endo$mean_beta))
model <- update(model, data = beta_mat_sp_its_nonster_endo, method = 'ML')
t_beta_its_endo <- merge_fix_rand(drop1(model, test = 'Chisq')[-1,], model) 
t_beta_its_endo %>% kable
```

endophty into two subsets
```{r}
print('pathogenic endophytes')
model <- update(model, data = beta_mat_sp_its_nonster_endo_path, control = NULL)
t_beta_its_endo_path <- merge_fix_rand(drop1(model, test = 'Chisq')[-1,], model)
t_beta_its_endo_path %>% kable
```


```{r}
print('non-pathogenic endophytes')
model <- update(model, data = beta_mat_sp_its_nonster_endo_no_path, control = lmeControl(msMaxIter = 10000))
t_beta_its_endo_no_path <- merge_fix_rand(drop1(model, test = 'Chisq')[-1,], model)
t_beta_its_endo_no_path %>% kable
```


## sterilized soil
### 16s (bacteria)
```{r}
u<-gl(1,1,length(beta_mat_sp_16s_ster$mean_beta))
model <- update(model, data = beta_mat_sp_16s_ster, weights = NULL)
t_beta_16s_ster <- merge_fix_rand(drop1(model, test = 'Chisq')[-1,], model)  # table
t_beta_16s_ster %>% kable
```


### its (fungi)
```{r}
print('all fungi')
u<-gl(1,1,length(beta_mat_sp_its_ster$mean_beta))
model<-update(model, 
              control = NULL, weights = NULL, 
              data = beta_mat_sp_its_ster)
t_beta_its_ster <- merge_fix_rand(drop1(model, test = 'Chisq')[-1,], model) 
t_beta_its_ster %>% kable
```


after sterilization, the differences went away, but similar pattern.
```{r}
print('pathogens')
u<-gl(1,1,length(beta_mat_sp_its_ster_pathogen$mean_beta))
model<-update(model, 
              control = lmeControl(opt = 'optim'), weights = varIdent(form = ~1|combi_status),
              data = beta_mat_sp_its_ster_pathogen, )

t_beta_its_path_ster <- merge_fix_rand(drop1(model, test = 'Chisq')[-1,], model) 
t_beta_its_path_ster %>% kable
```




```{r}
# some beta diversity is 1, which can not be logit transformed, so I bounded them to 0.99999
beta_mat_sp_its_ster_endo$mean_beta[beta_mat_sp_its_ster_endo$mean_beta > 0.999999]<-0.99999
u <- gl(1,1,length(beta_mat_sp_its_ster_endo$mean_beta))
model <- update(model, data = beta_mat_sp_its_ster_endo, control = NULL)
t_beta_its_endo_ster <- merge_fix_rand(drop1(model, test = 'Chisq')[-1,], model) 
t_beta_its_endo_ster %>% kable
```


## Figures
### non-sterilized soil (Fig. 4)
**high mean_beta (red) means very different soil communities**


Species by species matrix, blue means soil communities are different, red means similar.\
The topleft corner is alien vs alien, bottomright native vs native, and the other alien vs native.
```{r, fig.width=7, fig.height= 6}
# pdf('./fig/cor_plot_sp_t.pdf', height = 6, width = 18/2.54, useDingbats=FALSE)
par(mfrow = c(2,4),xpd = T)
plot_cor(beta_mat_sp_16s_nonster,title = 'Bacteria')
# text
text(-0.5,13.5, 'own-alien', cex = 0.85, col = 'gray35')
segments(1.45, 13.5, 3.5, 13.5, col = 'gray35')
Arrows(3.5,13.5, 3.5, 12.8, arr.type="triangle", arr.length = 0.05, col = 'gray35')

text(-0.5, 10.8, 'own-native', cex = 0.85, col = 'gray35')
segments(1.7,10.8, 8.5, 10.8, col = 'gray35')
Arrows(8.5,10.8, 8.5, 11.2, arr.type="triangle", arr.length = 0.05, col = 'gray35')

text(1.5,6.5, 'alien-alien', cex = 0.85)
Arrows(1.7, 6.7, 3.3, 8.3, arr.type="triangle", arr.length = 0.05)
text(3.5, 4.5, 'native-alien', cex = 0.85)
Arrows(3.7, 4.7, 5.3, 6.3, arr.type="triangle", arr.length = 0.05)
text(6.5, 1.5, 'native-native', cex = 0.85)
Arrows(6.7, 1.7, 8.3, 3.3, arr.type="triangle", arr.length = 0.05)


# fungi
plot_cor(beta_mat_sp_its_nonster,title = 'All fungi', tag = 'b')
# color scale
rect(-2, 9 - 0.5*1:9, -2.8, 8.5 - 0.5*1:9, col =
       colorRampPalette(c("red","white","blue"))(200)[c(8:1*25,1)], border = NA)
rect(-2, 8.5, -2.8, 4, lwd = 0.5, border = 'gray39')
text(-1.8, 8.5, adj = c(0,NA), 'different', cex = 0.7)
text(-1.7, 4, adj = c(0, NA),  'similar', cex = 0.7)
Arrows(- 1.3,7.9, -1.3, 4.6, arr.type="triangle", arr.length = 0.05, code = 3, lwd = 0.5) # connect different and similar


# pathogen
plot_cor(beta_mat_sp_its_nonster_pathogen,title = 'Fungal pathogens', tag = 'c')

# endophytes
plot_cor(beta_mat_sp_its_nonster_endo,title = 'Fungal endophytes', tag = 'd')

# forest plots
par(mar = c(15, 5.2, 0, 0.2), xpd = T)
forest_beta_t(title = '', ylab = T)
segments(0.42, 5.5, 0.42, 4.5, lwd = 0.75)
segments(0.47, 3,   0.47, 1,   lwd = 0.75)
segments(0.415, 2,   0.465, 2, lwd = 0.75)
brackets(0.415, 5,   0.415, 2,   type = 4, h = -0.01, ticks = NA, lwd = 0.75)
text(0.415-0.02, 3.5, '*')

forest_beta_t(beta_mat = beta_mat_sp_its_nonster, title = '', xaxt = c(0.6, 0.7, 0.8), xlim = c(0.58,0.83),
            tag = 'f')
brackets(0.79, 3, 0.79, 2,   type = 4, h = 0.01, ticks = NA, lwd = 0.75)
text(0.815, 2.5, '*')
# brackets(0.65, 2, 0.65, 1,   type = 4, h = -0.01, ticks = NA, lwd = 0.75)
# text(0.625, 1.5, '\u2020')

forest_beta_t(beta_mat = beta_mat_sp_its_nonster_pathogen, title = '', xaxt = c(0.6, 0.7, 0.8),xlim = c(0.58,0.83),
            tag = 'g')
forest_beta_t(beta_mat = beta_mat_sp_its_nonster_endo, title = '', xaxt = c(0.5, 0.6, 0.7),xlim = c(0.48,0.75),
            tag = 'h')
brackets(0.72, 3, 0.72, 2,   type = 4, h = 0.01, ticks = NA, lwd = 0.75)
text(0.745, 2.5, '*')
brackets(0.55, 2, 0.55, 1,   type = 4, h = -0.01, ticks = NA, lwd = 0.75)
text(0.525, 1.5, '*')
brackets(0.52, 5.5, 0.52, 4.5,   type = 4, h = -0.01, ticks = NA, lwd = 0.75)
text(0.495, 5, '*')

# dev.off()
```


### two subsets of endophytes (Extended Figure 6a-d)
```{r, , fig.width= 6, fig.height= 5.5}
# pdf('./fig/cor_plot_sp_endo_t.pdf',height = 6, width = 17/2.54,  useDingbats=FALSE)
par(mfrow = c(2,3),xpd = T)
plot_cor(beta_mat_sp_its_nonster_endo_path,title = 'Pathogenic endophytes')
plot_cor(beta_mat_sp_its_nonster_endo_no_path,title = 'Non-pathogenic endophytes', tag = 'b')
plot.new()
par(mar = c(12, 4.8, 0, 0.25), xpd = T)
forest_beta_t(beta_mat = beta_mat_sp_its_nonster_endo_path, title = '', ylab = T, tag = 'c',
               xaxt = c(0.5, 0.6, 0.7, 0.8), xlim = c(0.48,0.8))

brackets(0.55, 2, 0.55, 1,   type = 4, h = -0.01, ticks = NA, lwd = 0.75)
text(0.525, 1.5, '*')
brackets(0.52, 5.5, 0.52, 4.5,   type = 4, h = -0.01, ticks = NA, lwd = 0.75)
text(0.495, 5, '*')

forest_beta_t(beta_mat = beta_mat_sp_its_nonster_endo_no_path, title = '', 
              xaxt = c(0.5, 0.6, 0.7), xlim = c(0.48,0.72),tag = 'd')

brackets(0.65, 3, 0.65, 2,   type = 4, h = 0.01, ticks = NA, lwd = 0.75)
text(0.67, 2.5, '*')
brackets(0.5, 5.5, 0.5, 4.5,   type = 4, h = -0.01, ticks = NA, lwd = 0.75)
text(0.48, 5, '*')
# dev.off()
```




### sterilized soils (Extended Fig. 5)
```{r, fig.width=7, fig.height= 6}
# pdf('./fig/cor_plot_sp_ster_t.pdf',height = 6, width = 18/2.54)
par(mfrow = c(2,4),xpd = T)
plot_cor(beta_mat_sp_16s_ster,title = 'Bacteria')
plot_cor(beta_mat_sp_its_ster,title = 'All fungi', tag = 'b')
plot_cor(beta_mat_sp_its_ster_pathogen,title = 'Fungal pathogens', tag = 'c')
plot_cor(beta_mat_sp_its_ster_endo,title = 'Fungal endophytes', tag = 'd')

par(mar = c(15, 4.8, 0, 0.25), xpd = T)
forest_beta_t(beta_mat = beta_mat_sp_16s_ster, title = '', ylab = T, tag = 'e')
forest_beta_t(beta_mat = beta_mat_sp_its_ster, title = '', 
              xaxt = c(0.5, 0.6, 0.7, 0.8, 0.9), xlim = c(0.48,0.92), tag = 'f')
forest_beta_t(beta_mat = beta_mat_sp_its_ster_pathogen, title = '', 
              xaxt = c(0.5, 0.6, 0.7, 0.8),xlim = c(0.48,0.82), tag = 'g')
forest_beta_t(beta_mat = beta_mat_sp_its_ster_endo, title = '', 
              xaxt = c(0.1*6:10),xlim = c(0.58,1), tag = 'k')
# dev.off()
```

export tables
```{r}
# write.csv(cbind(t_beta_16s, t_beta_its, t_beta_its_path, t_beta_its_endo,
#                 t_beta_16s_ster, t_beta_its_ster, t_beta_its_path_ster, t_beta_its_endo_ster),
#           './table/table_beta.csv')
# 
# write.csv(cbind( t_beta_its_endo_path,t_beta_its_endo_no_path),
#           './table/table_beta_two_endo.csv')
```




# alpha diversity and abundance
**This section aims to test whether present of plants and the origin of the plants affect the alpha diversity (species richness and Shannon diversity) of soil microbes and abundance of fungal pathogens.**\

Note that, the abundance is 'relative' abundance, so it cannot be applied to whole bacteria and fungi.
```{r}
#get diversity for each species
alpha_16s <- alpha_div(otu_table_16s,sam_table_16s)
alpha_its <- alpha_div(otu_table_its,sam_table_its)

# its different functional groups
its_func_sub <- its_func %>% 
  filter(OTU_ID %in% as.character(colnames(otu_table_its)))

otu_table_its_pathogen <- otu_table_its[,as.character(its_func_sub$OTU_ID[its_func_sub[,'pathogen'] == 'pathogen'])]
alpha_its_pathogen <-alpha_div(otu_table_its_pathogen, sam_table_its)
```





## non-sterilized soil
Overall, aliens and natives did not differ in diversity of fungi and bacteria
```{r}
m_16s_sr <-lme((sr)~ T1_empty + T2_alien_native + trans_date_c,
               random = ~ 1|family/species,
               data = alpha_16s %>% filter(ster.not =='nonster'),
               # weights = varIdent(form = ~1|species), # weight did not improve the model
               method = 'ML'
              )
m_16s_sha <-update(m_16s_sr,shannon ~., control = lmeControl(msMaxIter = 1000))


m_its_sr  <- update(m_16s_sr,.~., data = alpha_its %>% filter(ster.not =='nonster'))
m_its_sha <- update(m_its_sr,shannon ~.)

m_its_pathogen_sr <- update(m_its_sr, data = alpha_its_pathogen %>% filter(ster.not == 'nonster'))
m_its_pathogen_sha <-update(m_its_pathogen_sr,shannon~.)


```


```{r}
drop1_div <- function(model){
  r1 <- drop1(model, test = 'Chisq')[-c(1,4),]
  r2 <- drop1(update(model,.~trans_date_c), test = 'Chisq')[-1,]
  return(merge_fix_rand(rbind(r2,r1), model, nest = T))
  
}
t_div_nonster <- lapply(list(m_16s_sr, m_16s_sha,
                             m_its_sr, m_its_sha,
                             m_its_pathogen_sr, m_its_pathogen_sha),
                        drop1_div)
```

### diversity of 16s (bacteria)
```{r}
print('Species richness');  t_div_nonster[[1]] %>% kable
print('Shannon diversity'); t_div_nonster[[2]] %>% kable
```

### diversity of its (fungi)
all fungi
```{r}
print('Species richness');  t_div_nonster[[3]] %>% kable
print('Shannon diversity'); t_div_nonster[[4]] %>% kable
```

fungal pathogens
```{r}
print('Species richness');  t_div_nonster[[5]] %>% kable
print('Shannon diversity'); t_div_nonster[[6]] %>% kable
```

### abundance of fungal pathogens

overview in both sterilized and non-sterilized soils
```{r} 
print(paste('total diversity = ',length(otu_table_its)))
rel_abun(otu_table_its,its_func,'assigned')
rel_abun(otu_table_its,its_func,'pathogen')
rel_abun(otu_table_its,its_func,'AMF')
rel_abun(otu_table_its,its_func,'endo')
rel_abun(otu_table_its,its_func,'endo_path')
rel_abun(otu_table_its,its_func,'endo_no_path')
```

only nonster
```{r}
print(paste('total diversity = ',length(otu_table_its_nonster)))
rel_abun(otu_table_its_nonster,its_func_nonster, 'assigned')
rel_abun(otu_table_its_nonster,its_func_nonster, 'pathogen')
rel_abun(otu_table_its_nonster,its_func_nonster, 'AMF')
rel_abun(otu_table_its_nonster,its_func_nonster, 'endo')
rel_abun(otu_table_its_nonster,its_func_nonster, 'endo_path')
rel_abun(otu_table_its_nonster,its_func_nonster, 'endo_no_path')
```

```{r}
abun_its_pathogen <- abun_by_pot(otu_table_its_pathogen,info_its)
```



models
```{r}
# not sterilized
model  <- lme(logit(rel_abu)~ T1_empty + T2_alien_native + trans_date_c, random = ~1|family/species, 
              weights = varIdent(form = ~1|species),
              data = abun_its_pathogen %>% filter(ster.not == 'nonster'),method = 'ML') 
# sterilized
model2 <- update(model, .~. + trans_date_c, data = abun_its_pathogen %>% filter(ster.not == 'ster'))

t_abundance <- lapply(list(model, model2), drop1_div)
```

```{r}
t_abundance[[1]] %>% kable()
```



## sterilized
```{r}
m_16s_sr<-lme((sr)~ T1_empty + T2_alien_native + trans_date_c,random = ~ 1|family/species, 
              data = alpha_16s %>% filter(ster.not =='ster'),
              # weights = varIdent(form = ~1|species), # weight did not improve the model
              method = 'ML')
m_16s_sha<-update(m_16s_sr,shannon ~., control = lmeControl(msMaxIter = 1000))


m_its_sr  <- update(m_16s_sr,.~., data = alpha_its %>% filter(ster.not =='ster'))
m_its_sha <- update(m_its_sr,shannon ~.)


m_its_pathogen_sr <- update(m_its_sr, data = alpha_its_pathogen %>% filter(ster.not == 'ster'))
m_its_pathogen_sha <-update(m_its_pathogen_sr, shannon~.)


t_div_ster <- lapply(list(m_16s_sr, m_16s_sha,
                             m_its_sr, m_its_sha,
                             m_its_pathogen_sr, m_its_pathogen_sha),
                        drop1_div)
```

### diversity of 16s (bacteria)
```{r}
print('Species richness');  t_div_ster[[1]] %>% kable
print('Shannon diversity'); t_div_ster[[2]] %>% kable
```


### diversity of its (fungi)
all fungi
```{r}
print('Species richness');  t_div_ster[[3]] %>% kable
print('Shannon diversity'); t_div_ster[[4]] %>% kable
```

fungal pathogens
```{r}
print('Species richness');  t_div_ster[[5]] %>% kable
print('Shannon diversity'); t_div_ster[[6]] %>% kable
```


### abundance of fungal pathogens
```{r}
t_abundance[[2]] %>% kable()
```


## Figures (Extended Fig. 3)
```{r alpha_fig}
# -------------diversity------------#
# bacteria
p_alpha_16s_sr   <- boxplot_div(table_alpha = alpha_16s, title = 'Bacteria', tag = 'a')
p_alpha_16s_shan <- boxplot_div(table_alpha = alpha_16s, title = '', 
                  y = 'shannon', ylab = 'Shannon diversity', tag = 'd')

# all fungi
p_alpha_its_sr   <- boxplot_div(table_alpha = alpha_its, title = 'All fungi', tag = 'b', ylab = ' ')
p_alpha_its_shan <- boxplot_div(table_alpha = alpha_its, title = '', 
                  y = 'shannon', ylab = ' ', tag = 'e')

# fungal pathogens
p_alpha_its_pathogen_sr <- boxplot_div(table_alpha = alpha_its_pathogen, title = 'Fungal pathogens', tag = 'c', ylab = ' ')
p_alpha_its_pathogen_shan <- boxplot_div(table_alpha = alpha_its_pathogen, title = '', y = 'shannon', ylab = ' ', tag = 'f')

# --------abundance -------#
p1 <- boxplot_div(abun_its_pathogen, y = 'rel_abu', title = 'Fungal pathogens',ylab = 'Relative abundance', tag = 'g')
```


```{r}
# ggsave('./fig/alpha.pdf', height = 7, width = 18/2.54,
#        grid.arrange(p_alpha_16s_sr,   p_alpha_its_sr,   p_alpha_its_pathogen_sr,
#                     p_alpha_16s_shan, p_alpha_its_shan, p_alpha_its_pathogen_shan,
#                     p1, 
#                     layout_matrix = matrix(c(1:6, NA, NA, 7), nrow =3, ncol = 3, byrow = T)))
```
```{r,fig.height = 7, fig.width= 7}
grid.arrange(p_alpha_16s_sr,   p_alpha_its_sr,   p_alpha_its_pathogen_sr,
             p_alpha_16s_shan, p_alpha_its_shan, p_alpha_its_pathogen_shan,
             p1, 
             layout_matrix = matrix(c(1:6, NA, NA, 7), nrow =3, ncol = 3, byrow = T))
```


export tables
```{r}
# write.csv(t_div_nonster, './table/t_div_nonster.csv') # diversity (not sterilized)
# write.csv(t_div_ster,    './table/t_div_ster.csv') # diversity (sterilized)
# 
# write.csv(t_abundance, './table/t_abundance.csv') # abundance
```





# soil-legacy effect ~ soil communities
**This section aims to test which aspect of soil communities (alpha) can explain the soil-legacy effect (beta).**\
I tested two aspects: enemy release (diversity and abundance of soil microbes) and novel interaction (dissimilarity between soil communities).


data prep


```{r}
beta_16s_combi<-beta_mat_sp_16s_nonster%>%
  mutate(combi = ifelse(sp1 < sp2,paste(sp1,sp2),paste(sp2,sp1)))%>%# get a order, sp1 is the soil species, and sp2 is the test species
  ungroup()%>%
  dplyr::select(combi,mean_beta)%>%
  dplyr::rename(beta_16s = mean_beta)
beta_its_combi<-beta_mat_sp_its_nonster%>%
  mutate(combi = ifelse(sp1 < sp2,paste(sp1,sp2),paste(sp2,sp1)))%>%# get a order
  ungroup()%>%
  dplyr::select(combi,mean_beta)%>%
  dplyr::rename(beta_its = mean_beta)
beta_its_pathogen_combi<-beta_mat_sp_its_nonster_pathogen%>%
  mutate(combi = ifelse(sp1 < sp2,paste(sp1,sp2),paste(sp2,sp1)))%>%# get a order
  ungroup()%>%
  dplyr::select(combi,mean_beta)%>%
  dplyr::rename(beta_its_pathogen = mean_beta)
beta_its_endo_combi<-beta_mat_sp_its_nonster_endo%>%
  mutate(combi = ifelse(sp1 < sp2,paste(sp1,sp2),paste(sp2,sp1)))%>%# get a order
  ungroup()%>%
  dplyr::select(combi,mean_beta)%>%
  dplyr::rename(beta_its_endo = mean_beta)
beta_its_endo_path_combi<-beta_mat_sp_its_nonster_endo_path%>%
  mutate(combi = ifelse(sp1 < sp2,paste(sp1,sp2),paste(sp2,sp1)))%>%# get a order
  ungroup()%>%
  dplyr::select(combi,mean_beta)%>%
  dplyr::rename(beta_its_endo_path = mean_beta)
beta_its_endo_no_path_combi<-beta_mat_sp_its_nonster_endo_no_path%>%
  mutate(combi = ifelse(sp1 < sp2,paste(sp1,sp2),paste(sp2,sp1)))%>%# get a order
  ungroup()%>%
  dplyr::select(combi,mean_beta)%>%
  dplyr::rename(beta_its_endo_no_path = mean_beta)



# abundance of pathogen
abun_path <- abun_its_pathogen %>% 
  filter(ster.not == 'nonster') %>% 
  group_by(species) %>% 
  summarise(abun_path = mean(rel_abu))

# alpha diversity
div_path <- alpha_its_pathogen %>% 
  filter(ster.not == 'nonster') %>% 
  dplyr::select(sr, shannon, species) %>% 
  group_by(species) %>% 
  summarise(div_path_sr = mean(sr),
            div_path_shan = mean(shannon))

div_its <- alpha_its %>% 
  filter(ster.not == 'nonster') %>% 
  dplyr::select(sr, shannon, species) %>% 
  group_by(species) %>% 
  summarise(div_its_sr = mean(sr),
            div_its_shan = mean(shannon))

div_16s <- alpha_16s %>% 
  filter(ster.not == 'nonster') %>% 
  dplyr::select(sr, shannon, species) %>% 
  group_by(species) %>% 
  summarise(div_16s_sr = mean(sr),
            div_16s_shan = mean(shannon))
```




data merge. The `legacy_alone` is the soil-legacy effect (beta_alone in Fig. 1), which is calculated from the test plants that were grown alone\

sp_p1: soil species\
sp_p2: test species
```{r}
dat_legacy_alone <- dat_legacy_alone %>% 
  mutate(sp_p1 = as.character(sp_p1),sp_p2 = as.character(sp_p2))%>%
  mutate(combi = ifelse(sp_p1 < sp_p2, paste(sp_p1,sp_p2), paste(sp_p2,sp_p1))) %>% # unique set
  left_join(beta_16s_combi) %>%
  left_join(beta_its_combi) %>%
  left_join(beta_its_pathogen_combi) %>%
  left_join(beta_its_endo_combi) %>%
  left_join(beta_its_endo_path_combi) %>%
  left_join(beta_its_endo_no_path_combi) %>% 
  left_join(abun_path, by = c('sp_p1' = 'species')) %>% 
  left_join(div_path,  by = c('sp_p1' = 'species')) %>% 
  left_join(div_16s,   by = c('sp_p1' = 'species')) %>% 
  left_join(div_its,   by = c('sp_p1' = 'species')) %>% 
  mutate(combi_status = ifelse(origin_p1 == 'native' & origin_p2 == 'native','native-native',
                               ifelse(origin_p1 == 'alien' & origin_p2 == 'alien','alien-alien','native-alien'))) %>% 
  mutate(intra_inter = ifelse(sp_p1 != sp_p2, 0 , 1))
```


The `legacy_stren` is also soil-legacy effect. But it is the beta_intra and beta_inter in Fig. 1), which describes the effect of soil species on strength of competition in test plants.\

sp_p2_comp: competitor test species
```{r}
dat_legacy_stren <- dat_legacy_stren %>% 
  mutate(sp_p1 = as.character(sp_p1), sp_p2 = as.character(sp_p2), sp_p2_comp = as.character(sp_p2_comp)) %>% 
  mutate(combi_p1_target = ifelse(sp_p1 < sp_p2,      paste(sp_p1, sp_p2),      paste(sp_p2, sp_p1)),
         combi_p1_comp   = ifelse(sp_p1 < sp_p2_comp, paste(sp_p1, sp_p2_comp), paste(sp_p2_comp, sp_p1)),
         combi_p2        = ifelse(sp_p2 < sp_p2_comp, paste(sp_p2, sp_p2_comp), paste(sp_p2_comp, sp_p2))) %>% 
  left_join(abun_path, by = c('sp_p1' = 'species')) %>% 
  left_join(div_path,  by = c('sp_p1' = 'species')) %>% 
  left_join(div_16s,   by = c('sp_p1' = 'species')) %>% 
  left_join(div_its,   by = c('sp_p1' = 'species')) %>% 
  # as we have three individual in each system, we will have three beta diversities for each type of beta diversity
  # i.e. beta diversity between soil (p1) individual and test focal individual (_p1_target), between soil individual and test competitor individual (_p1_comp), and between test individuals (_p2)
  left_join(beta_16s_combi, by = c('combi_p1_target' = 'combi')) %>% rename(beta_16s_p1_target = beta_16s) %>% 
  left_join(beta_16s_combi, by = c('combi_p1_comp'   = 'combi')) %>% rename(beta_16s_p1_comp   = beta_16s) %>% 
  left_join(beta_16s_combi, by = c('combi_p2'        = 'combi')) %>% rename(beta_16s_p2        = beta_16s) %>% 
  
  left_join(beta_its_combi, by = c('combi_p1_target' = 'combi')) %>% rename(beta_its_p1_target = beta_its) %>% 
  left_join(beta_its_combi, by = c('combi_p1_comp'   = 'combi')) %>% rename(beta_its_p1_comp = beta_its) %>% 
  left_join(beta_its_combi, by = c('combi_p2'        = 'combi')) %>% rename(beta_its_p2 = beta_its) %>% 

  left_join(beta_its_pathogen_combi, by = c('combi_p1_target' = 'combi')) %>% rename(beta_its_path_p1_target = beta_its_pathogen) %>% 
  left_join(beta_its_pathogen_combi, by = c('combi_p1_comp'   = 'combi')) %>% rename(beta_its_path_p1_comp = beta_its_pathogen) %>% 
  left_join(beta_its_pathogen_combi, by = c('combi_p2'        = 'combi')) %>% rename(beta_its_path_p2 = beta_its_pathogen) %>% 

  left_join(beta_its_endo_combi, by = c('combi_p1_target' = 'combi')) %>% rename(beta_its_endo_p1_target = beta_its_endo) %>% 
  left_join(beta_its_endo_combi, by = c('combi_p1_comp'   = 'combi')) %>% rename(beta_its_endo_p1_comp = beta_its_endo) %>% 
  left_join(beta_its_endo_combi, by = c('combi_p2'        = 'combi')) %>% rename(beta_its_endo_p2 = beta_its_endo) %>%
  
  # interspecific competition or inter
  mutate(intra_inter = ifelse(sp_p2 != sp_p2_comp, 'inter', 'intra'))
```


## enemy release
### alone (Extended Fig. 4)
legacy ~ abundance of fungal pathogens
```{r}
u<-gl(1,1,length(dat_legacy_alone$biomass))
model<-lme(legacy_alone ~ logit(abun_path) * origin_p2,
           random=list(u=pdBlocked(list(
            pdIdent(form=~family_p1-1),
            pdIdent(form=~sp_p1-1),
            pdIdent(form=~family-1),
            pdIdent(form=~sp_p2-1)
            ))),
          data=dat_legacy_alone,
          weights = varComb(varIdent(form = ~1|sp_p2),
                            varIdent(form = ~ 1|sp_p1)),
          control = lmeControl(msMaxIter = 1000),
          method = "ML")
r1 <- drop1(model, test ='Chisq')[-1,]
r2 <- drop1(update(model,.~.- logit(abun_path) : origin_p2), test ='Chisq')[-1,]

t_legacy_ab_path <- merge_fix_rand(rbind(r2,r1), model)
t_legacy_ab_path %>% kable
```



legacy ~ diversity
16s_species_richness
```{r}
model<-lme(legacy_alone ~ div_16s_sr * origin_p2,
           random=list(u=pdBlocked(list(
            pdIdent(form=~family_p1-1),
            pdIdent(form=~sp_p1-1),
            pdIdent(form=~family-1),
            pdIdent(form=~sp_p2-1)
            ))),
          data=dat_legacy_alone,
          weights = varComb(varIdent(form = ~1|sp_p2),
                            varIdent(form = ~ 1|sp_p1)),
          control = lmeControl(msMaxIter = 1000),
          method = "ML")
r1 <- drop1(model, test ='Chisq')[-1,]
r2 <- drop1(update(model,.~.- div_16s_sr : origin_p2), test ='Chisq')[-1,]
t_legacy_sr_16s <- merge_fix_rand(rbind(r2,r1), model)
t_legacy_sr_16s %>% kable
```


16s_shannon
```{r}
model<- update(model, .~ div_16s_shan * origin_p2)
r1 <- drop1(model, test ='Chisq')[-1,]
r2 <- drop1(update(model,.~.- div_16s_shan : origin_p2), test ='Chisq')[-1,]
t_legacy_shan_16s <- merge_fix_rand(rbind(r2,r1), model)
t_legacy_shan_16s %>% kable
```

its_species_richness
```{r}
model<- update(model, .~ div_its_sr * origin_p2)
r1 <- drop1(model, test ='Chisq')[-1,]
r2 <- drop1(update(model,.~.- div_its_sr : origin_p2), test ='Chisq')[-1,]
t_legacy_sr_its <- merge_fix_rand(rbind(r2,r1), model)
t_legacy_sr_its %>% kable
```


its_shannon_diversity
```{r}
model<- update(model, .~ div_its_shan * origin_p2)
r1 <- drop1(model, test ='Chisq')[-1,]
r2 <- drop1(update(model,.~.- div_its_shan : origin_p2), test ='Chisq')[-1,]
t_legacy_shan_its <- merge_fix_rand(rbind(r2,r1), model)
t_legacy_shan_its %>% kable
```



pathogen_species_richness
```{r}
model1<- update(model, .~ div_path_sr * origin_p2)
r1 <- drop1(model1, test ='Chisq')[-1,]
r2 <- drop1(update(model1,.~.- div_path_sr : origin_p2), test ='Chisq')[-1,]
rbind(r2,r1) %>% lrt.table()
t_legacy_sr_path <- merge_fix_rand(rbind(r2,r1), model1)
```

pathogen_shannon_diversity
```{r}
model2 <- update(model, .~ div_path_shan * origin_p2)
r1 <- drop1(model2, test ='Chisq')[-1,]
r2 <- drop1(update(model2,.~.- div_path_shan : origin_p2), test ='Chisq')[-1,]
rbind(r2,r1) %>% lrt.table()

t_legacy_shan_path <- merge_fix_rand(rbind(r2,r1), model2)
```


export table
```{r}
# write.csv(cbind(t_legacy_ab_path,
#                 t_legacy_sr_16s,   t_legacy_sr_its,   t_legacy_sr_path, 
#                 t_legacy_shan_16s, t_legacy_shan_its, t_legacy_shan_path),
#           './table/legacy_ab_div.csv')
```




```{r}
ylab = expression(paste('Soil legacy effect (', italic(beta['alone']), ')','\n', sep = ''))
p1 <- dat_legacy_alone %>% 
  ggplot() + aes(x = div_16s_sr, y = legacy_alone, col = origin_p2) + 
  geom_point(size = 0.8) + labs(x = '\nSpecies richness', y = ylab, 
                                title = 'Bacteria', tag = 'a') + 
  theme + scale_x_continuous(limits = c(3000, 4000),
                             breaks=pretty_breaks(n = 3)) +
  scale_color_manual(values = t_col(col_fig)) + 
  # legend
  theme(legend.position = c(0.72,0.2),
        legend.text = element_text(size = 7),
        legend.key.height = unit(0.6,'line'))

p2 <- dat_legacy_alone %>% 
  ggplot() + aes(x = div_its_sr, y = legacy_alone, col = origin_p2) + 
  geom_point(size = 0.8) + labs(x = '\nSpecies richness', y = ' ', 
                                title = 'All fungi', tag = 'b') + 
  theme + scale_x_continuous(breaks=pretty_breaks(n = 3)) +
  scale_color_manual(values = t_col(col_fig))

p3 <- dat_legacy_alone %>% 
  ggplot() + aes(x = div_16s_shan, y = legacy_alone, col = origin_p2) + 
  geom_point(size = 0.8) + labs(x = '\nShannon diversity', y = ylab, 
                                title = ' ', tag = 'd') + 
  theme + scale_x_continuous(limits = c(6.6, 7.2), breaks=pretty_breaks(n = 3)) +
  scale_color_manual(values = t_col(col_fig))
p4 <- dat_legacy_alone %>% 
  ggplot() + aes(x = div_its_shan, y = legacy_alone, col = origin_p2) + 
  geom_point(size = 0.8) + labs(x = '\nShannon diversity', y = ' ', 
                                title = ' ', tag = 'e') + 
  theme + scale_x_continuous(breaks=pretty_breaks(n = 3)) +
  scale_color_manual(values = t_col(col_fig))
```


```{r}
p5 <- dat_legacy_alone %>% 
  ggplot() + aes(x = logit(abun_path), y = legacy_alone, col = origin_p2) + 
  geom_point(size = 0.8) + labs(x = 'Relative abundance', y = ylab, 
                                title = ' ', tag = 'g') + 
  theme + scale_x_continuous(# limits = c(0.425, 0.55),
                             breaks=pretty_breaks(n = 3)) +
  scale_color_manual(values = t_col(col_fig))

# diversity
p6 <- dat_legacy_alone %>% 
  ggplot() + aes(x = div_path_sr, y = legacy_alone, col = origin_p2) + 
  geom_point(size = 0.8) + labs(x = 'Species richness', y = ' ', title = 'Fungal pathogens', tag = 'c') + 
  theme + scale_x_continuous(# limits = c(0.425, 0.55),
                             breaks=pretty_breaks(n = 3)) +
  scale_color_manual(values = t_col(col_fig))



p7 <- dat_legacy_alone %>% 
  ggplot() + aes(x = div_path_shan, y = legacy_alone, col = origin_p2) + 
  geom_point(size = 0.8) + labs(x = 'Shannon diversity', y = ' ', title = ' ', tag = 'f') + 
  theme + scale_x_continuous(# limits = c(0.425, 0.55),
                             breaks=pretty_breaks(n = 4)) +
  scale_color_manual(values = t_col(col_fig))
```

```{r}
# ggsave('./fig/legacy~div.pdf', height = 6, width = 14/2.54,
#        grid.arrange(p1,   p2,   p6,
#                     p3, p4, p7,
#                     p5,
#                     layout_matrix = matrix(c(1:6, NA, NA, 7), nrow =3, ncol = 3, byrow = T)))
```


### strength of competition (Supplementary Fig.4)

```{r}
u<-gl(1,1,length(dat_legacy_stren$legacy_stren))
```

16s species richness
```{r}
model <- lme(legacy_stren ~ intra_inter + div_16s_sr * origin_p2 ,
           random=list(u=pdBlocked(list(pdIdent(form=~family_p1-1), pdIdent(form=~sp_p1-1),
                                        pdIdent(form=~family-1),    pdIdent(form=~sp_p2-1),
                                        pdIdent(form=~family_comp-1),pdIdent(form=~sp_p2_comp-1)))),
            data=dat_legacy_stren,
            weights = varComb(varIdent(form = ~ 1|sp_p2),
                              varIdent(form = ~ 1|sp_p1),
                              varIdent(form = ~ 1|sp_p2_comp)),
            control = lmeControl(msMaxIter = 1000),
            method = "ML")
```
```{r}
r1 <- drop1(model, test ='Chisq')[-c(1:2),]
r2 <- drop1(update(model,.~.- div_16s_sr : origin_p2), test ='Chisq')[-1,]
t_legacy_sr_16s_stren <- merge_fix_rand(rbind(r2,r1), model)
t_legacy_sr_16s_stren %>% kable
```

16s shannon diversity
```{r}
model2 <- update(model, .~ intra_inter + div_16s_shan * origin_p2 )
r1 <- drop1(model2, test ='Chisq')[-c(1:2),]
r2 <- drop1(update(model2,.~.- div_16s_shan : origin_p2), test ='Chisq')[-1,]
t_legacy_sh_16s_stren <- merge_fix_rand(rbind(r2,r1), model2)
t_legacy_sh_16s_stren %>% kable
```


its species richness
```{r}
model3 <- update(model, .~ intra_inter + div_its_sr * origin_p2 )
r1 <- drop1(model3, test ='Chisq')[-c(1:2),]
r2 <- drop1(update(model3,.~.- div_its_sr : origin_p2), test ='Chisq')[-1,]
t_legacy_sr_its_stren <- merge_fix_rand(rbind(r2,r1), model3)
t_legacy_sr_its_stren %>% kable
```

its shannon diversity
```{r}
model4 <- update(model, .~ intra_inter + div_its_shan * origin_p2 )
r1 <- drop1(model4, test ='Chisq')[-c(1:2),]
r2 <- drop1(update(model4,.~.- div_its_shan : origin_p2), test ='Chisq')[-1,]
t_legacy_sh_its_stren <- merge_fix_rand(rbind(r2,r1), model4)
t_legacy_sh_its_stren %>% kable
```



fungal pathogens
```{r}
print('abundance')
model5 <- update(model, .~ intra_inter + abun_path * origin_p2 )
r1 <- drop1(model5, test ='Chisq')[-c(1:2),]
r2 <- drop1(update(model5,.~.- abun_path : origin_p2), test ='Chisq')[-1,]
t_legacy_ab_path_stren <- merge_fix_rand(rbind(r2,r1), model4)
t_legacy_ab_path_stren %>% kable

print('species richness')
model6 <- update(model, .~ intra_inter + div_path_sr * origin_p2 )
r1 <- drop1(model6, test ='Chisq')[-c(1:2),]
r2 <- drop1(update(model6,.~.- div_path_sr : origin_p2), test ='Chisq')[-1,]
t_legacy_sr_path_stren <- merge_fix_rand(rbind(r2,r1), model4)
t_legacy_sr_path_stren %>% kable

print('shannon diversity')
model7 <- update(model, .~ intra_inter + div_path_shan * origin_p2 )
r1 <- drop1(model7, test ='Chisq')[-c(1:2),]
r2 <- drop1(update(model7,.~.- div_path_shan : origin_p2), test ='Chisq')[-1,]
t_legacy_sh_path_stren <- merge_fix_rand(rbind(r2,r1), model4)
t_legacy_sh_path_stren %>% kable
```


```{r}
# write.csv(rbind(t_legacy_sr_16s_stren, t_legacy_sh_16s_stren, t_legacy_sr_its_stren, t_legacy_sh_its_stren), './table/legacy_div_stren.csv')
# write.csv(rbind(t_legacy_ab_path_stren, t_legacy_sr_path_stren, t_legacy_sh_path_stren), './table/legacy_ab_div_stren.csv')
```


```{r}
model2 <-update(model2, .~. - intra_inter, method = 'REML')
```

```{r}
# figure bacteria and fungi
ylab = expression(paste('Soil legacy effect (', italic(beta['inter_intra']), ')','\n', sep = ''))
p1 <- dat_legacy_stren %>% 
  ggplot() + aes(x = div_16s_sr, y = legacy_stren, col = origin_p2) + 
  geom_point(size = 0.8) + labs(x = '\nSpecies richness', y = ylab, 
                                title = 'Bacteria', tag = 'a') + 
  theme + scale_x_continuous(breaks=pretty_breaks(n = 3)) +
  scale_color_manual(values = t_col(col_fig)) +
  theme(legend.position = c(0.8,0.2),
        legend.text = element_text(size = 7),
        legend.key.height = unit(0.8,'line'))

p2 <- dat_legacy_stren %>% 
  ggplot() + aes(x = div_its_sr, y = legacy_stren, col = origin_p2) + 
  geom_point(size = 0.8) + labs(x = '\nSpecies richness', y = ' ', 
                                title = 'All fungi', tag = 'b') + 
  theme + scale_x_continuous(breaks=pretty_breaks(n = 4)) +
  scale_color_manual(values = t_col(col_fig))


p3 <- dat_legacy_stren %>% 
  ggplot() + aes(x = div_16s_shan, y = legacy_stren, col = origin_p2) + 
  geom_point(size = 0.8) + labs(x = '\nShannon diversity', y = ylab, 
                                title = ' ', tag = 'd') + 
  theme + scale_x_continuous(breaks=pretty_breaks(n = 4)) +
  scale_color_manual(values = t_col(col_fig)) +
  geom_abline(intercept = summary(model2)$coef$fixed[1], slope = summary(model2)$coef$fixed[2], col = col_fig[1]) +
  geom_abline(intercept = summary(model2)$coef$fixed[c(1,3)] %>% sum , slope = summary(model2)$coef$fixed[c(2,4)] %>% sum, col = col_fig[2]) 

p4 <- dat_legacy_stren %>% 
  ggplot() + aes(x = div_its_shan, y = legacy_stren, col = origin_p2) + 
  geom_point(size = 0.8) + labs(x = '\nShannon diversity', y = ' ', 
                                title = ' ', tag = 'e') + 
  theme + scale_x_continuous(breaks=pretty_breaks(n = 4)) +
  scale_color_manual(values = t_col(col_fig))
```



```{r}
model7 <-update(model7, .~. - intra_inter, method = 'REML')
```

figure fungal pathogens
```{r}

p5 <- dat_legacy_stren %>% 
  ggplot() + aes(x = abun_path, y = legacy_stren, col = origin_p2) + 
  geom_point(size = 0.8) + labs(x = '\nRelative abundance', y = ylab, 
                                title = ' ', tag = 'g') + 
  theme + scale_x_continuous(breaks=pretty_breaks(n = 4)) +
  scale_color_manual(values = t_col(col_fig))

p6 <- dat_legacy_stren %>% 
  ggplot() + aes(x = div_path_sr, y = legacy_stren, col = origin_p2) + 
  geom_point(size = 0.8) + labs(x = '\nSpecies richness', y = ' ', 
                                title = 'Fungal pathogens', tag = 'c') + 
  theme + scale_x_continuous(breaks=pretty_breaks(n = 3)) +
  scale_color_manual(values = t_col(col_fig))



p7 <- dat_legacy_stren %>% 
  ggplot() + aes(x = div_path_shan, y = legacy_stren, col = origin_p2) + 
  geom_point(size = 0.8) + labs(x = '\nShannon diversity', y = ' ', 
                                title = ' ', tag = 'f') + 
  theme + scale_x_continuous(breaks=pretty_breaks(n = 4)) +
  scale_color_manual(values = t_col(col_fig)) +
  geom_abline(intercept = summary(model7)$coef$fixed[1], slope = summary(model7)$coef$fixed[2], col = col_fig[1]) +
  geom_abline(intercept = summary(model7)$coef$fixed[c(1,3)] %>% sum , slope = summary(model7)$coef$fixed[c(2,4)] %>% sum, col = col_fig[2]) 
```


```{r}
# ggsave('./fig/alpha.pdf', height = 7, width = 18/2.54,
#        grid.arrange(p_alpha_16s_sr,   p_alpha_its_sr,   p_alpha_its_pathogen_sr,
#                     p_alpha_16s_shan, p_alpha_its_shan, p_alpha_its_pathogen_shan,
#                     p1, 
#                     layout_matrix = matrix(c(1:6, NA, NA, 7), nrow =3, ncol = 3, byrow = T)))
```

```{r,fig.height = 7, fig.width= 7}
grid.arrange(p1, p2, p6,
             p3, p4, p7,
             p5, layout_matrix = matrix(c(1:6, NA, NA, 7), nrow =3, ncol = 3, byrow = T))
      
```




# taxa
most abundant phyla in bacteria and fungi. (Supplementary Figure 2)

```{r}
reads_no <- sum(otu_table_16s[1,])
otu_table_16s_tax <- (otu_table_16s/reads_no) %>% 
  rownames_to_column('sample_id') %>% 
  gather('gene', 'abundance', -sample_id) %>% 
  left_join(tax_table_16s %>% rownames_to_column('gene')) %>% #join tax
  left_join(sam_table_16s) %>% 
  group_by(species,sample_id, Phylum, status, ster.not) %>% 
  summarise(abundance = sum(abundance)) %>% 
  group_by(species, Phylum, status, ster.not) %>% 
  summarise(`Relative abundance` = mean(abundance)) %>% 
  ungroup() %>% 
  mutate(status = factor(status, levels = c( 'non-conditioned', 'alien', 'native'))) %>% 
  arrange(ster.not, status, species)

otu_table_16s_tax$species <- factor(otu_table_16s_tax$species, levels = as.character(unique(otu_table_16s_tax$species))) # order by status
```


```{r}
reads_no <- sum(otu_table_its[1,])
otu_table_its_tax <- (otu_table_its/reads_no) %>% 
  rownames_to_column('sample_id') %>% 
  gather('gene', 'abundance', -sample_id) %>% 
  left_join(tax_table_its %>% rownames_to_column('gene')) %>% #join tax
  left_join(sam_table_its) %>% 
  group_by(species,sample_id, Phylum, species, status, ster.not) %>% 
  summarise(abundance = sum(abundance)) %>% 
  group_by(species, Phylum, species, status, ster.not) %>% 
  summarise(`Relative abundance` = mean(abundance)) %>% 
  ungroup() %>% 
  mutate(status = factor(status, levels = c( 'non-conditioned', 'alien', 'native'))) %>% 
  arrange(ster.not, status, species)

otu_table_its_tax$species <- factor(otu_table_its_tax$species, levels = as.character(unique(otu_table_its_tax$species)))
```


```{r}
# get top 5 abundant phyla
top <- (otu_table_16s_tax %>% 
  group_by(Phylum) %>% 
  summarise(mean = mean(`Relative abundance`)) %>% 
  arrange(-mean) %>% 
  top_n(5)) $ Phylum %>% as.character()
otu_table_16s_tax %>% 
  mutate(phylum_clean = ifelse(is.na(Phylum), 'not assigned',
                                ifelse(Phylum%in% top, as.character(Phylum),'others'))) %>% 
  mutate(phylum_clean = factor(phylum_clean, levels = c('not assigned', 'others', top[5:1]))) %>% 
  ggplot() + aes( x = species, y = `Relative abundance`, fill = phylum_clean) + geom_bar(position="stack", stat="identity") + 
  facet_wrap( ~ ster.not,labeller = labeller(ster.not = c('nonster' = 'Live', 'ster' = 'Sterilized'))) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.3, color = c('black',rep(col_fig[1],4), rep(col_fig[2],6)))) + 
  ggtitle('Bacteria') +
  scale_fill_manual(values = brewer.pal(8,'Set2')[c(8,6:1)]) + # filling colors
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), # remove background
        panel.background = element_blank(), axis.line = element_line(colour = "black"),
        legend.title = element_blank(),
        plot.title = element_text(hjust = 0.5))
# ggsave('./fig/sup_taxa_16s.pdf', plot = last_plot(), width = 6, height = 4)
```


```{r}
# get top abundant phyla
top <- (otu_table_its_tax %>% 
  group_by(Phylum) %>% 
  summarise(mean = mean(`Relative abundance`)) %>% 
  arrange(-mean) %>% 
  top_n(6)) $ Phylum %>% as.character()

otu_table_its_tax %>% 
  mutate(phylum_clean = ifelse(is.na(Phylum), 'not assigned',
                                ifelse(Phylum%in% top, as.character(Phylum),'others'))) %>%
  mutate(phylum_clean = sub('\\w__', '',phylum_clean)) %>% #remove 'w_'
  mutate(phylum_clean = factor(phylum_clean, levels = c('not assigned', 'others', 'Rozellomycota', 'Basidiomycota',
                                                        'Mortierellomycota', 'Chytridiomycota', 'Ascomycota'))) %>% 
  ggplot() + aes( x = species, y = `Relative abundance`, fill = phylum_clean) + geom_bar(position="stack", stat="identity") + 
  facet_wrap( ~ ster.not,labeller = labeller(ster.not = c('nonster' = 'Live', 'ster' = 'Sterilized'))) + 
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.3, color = c('black',rep(col_fig[1],4), rep(col_fig[2],6)))) +
  ggtitle('Fungi') + 
  scale_fill_manual(values = brewer.pal(8,'Set2')[c(8,6:1)]) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(), axis.line = element_line(colour = "black"),
        legend.title = element_blank(),
        plot.title = element_text(hjust = 0.5))

# ggsave('./fig/sup_taxa_its.pdf', plot = last_plot(), width = 6, height = 4)
```


# phylogenetic distance
Supplementary matrial 7
```{r tree}
# smith tree
tree <- read.tree('06tree.tre')
tree$tip.label[tree$tip.label == 'Scorzoneroides_autumnalis'] <- 'Leontodon_autumnalis'
pd <- cophenetic(tree) %>% 
  as.data.frame() %>% 
  rownames_to_column('sp1') %>% 
  gather(value = 'pd', key = 'sp2', -sp1) %>% 
  dplyr::left_join(info_its %>% dplyr::select(species, status) %>% distinct(), by = c('sp1' = 'species')) %>% 
  rename(status1 = status) %>% 
  dplyr::left_join(info_its %>% dplyr::select(species, status) %>% distinct(), by = c('sp2' = 'species')) %>% 
  rename(status2 = status)
```


Supplementary Figure 6 
```{r}
# pdf('./fig/tree.pdf', height = 6,width = 8)
par(xpd = T)
plot(tree, tip.color	= c(col_fig[1], col_fig[2],col_fig[1],col_fig[2], 
                         col_fig[2],col_fig[1],col_fig[2],col_fig[1],
                         col_fig[2],col_fig[2]))
axisPhylo()
text(60, -1,'Million year')
# dev.off()
```


phylogenetic distance ~ type (among aliens, among native, among natives)
```{r}
pd_test <- pd %>% 
  mutate(status1 = as.character(status1),
         status2 = as.character(status2)) %>% 
  mutate(combi = ifelse(status1 < status2, paste(status1, status2, sep = '-'), 
                        paste(status2,status1, sep = '-')),
         combi_sp = ifelse(sp1 < sp2, paste(sp1, sp2, sep = '-'), 
                        paste(sp2,sp1, sep = '-'))) %>% 
  dplyr::select(combi, combi_sp, pd) %>% 
  distinct() %>% 
  mutate(aa_vs_na = ifelse(combi == 'alien-alien', 1, 0),
         na_vs_nn = ifelse(combi =='native-native', 1, 0))
```

kruskal.test
```{r}
kruskal.test(pd ~ combi, data = pd_test)
```


combine pd and beta
```{r}
beta_pd_its_nonster <- beta_mat_sp_its_nonster %>% 
    left_join(pd)
beta_pd_its_nonster_pathogen <- beta_mat_sp_its_nonster_pathogen %>% 
    left_join(pd)
beta_pd_its_nonster_endo <- beta_mat_sp_its_nonster_endo %>% 
    left_join(pd)
beta_pd_16s_nonster <- beta_mat_sp_16s_nonster %>% 
    left_join(pd)
```



beta diversity (Bray-Curtis dissimilarity) ~ phylogenetic distance\
mantel test
```{r}
print('bacteria')
test<- dist_mat(beta_pd_16s_nonster)
mantel(as.dist(test[[1]]), as.dist(test[[2]]), na.rm = T)

print('Fungi')
test<- dist_mat()
mantel(as.dist(test[[1]]), as.dist(test[[2]]), na.rm = T)

print('pathogen')
test<- dist_mat(beta_pd_its_nonster_pathogen)
mantel(as.dist(test[[1]]), as.dist(test[[2]]), na.rm = T)

print('endo')
test<- dist_mat(beta_pd_its_nonster_endo)
mantel(as.dist(test[[1]]), as.dist(test[[2]]), na.rm = T)
```


# sessionInfo
```{r}
sessionInfo()
```



